trackintel/preprocessing/triplegs.py

```diff
@@ -113,30 +115,32 @@ def generate_trips(spts, tpls, gap_threshold=15):
 
     # delete intermediate activities
     # intermediate activities are activities that are surrounded by other activities
-    # therefor their are neither at the end, start, nor in a trip
+    # therefore they are neither at the end, at the start, nor within a trip
     first_activities, inter_activities, last_activities = _get_activity_masks(spts_tpls)
     spts_tpls["first_activity"] = first_activities
     spts_tpls["last_activity"] = last_activities  # trip may start here
     spts_tpls = spts_tpls[~inter_activities]  # no trip interaction
     spts_tpls["started_at_next"] = spts_tpls["started_at"].shift(-1)
 
     # conditions for new trip
-    # new user flag
+    # start new trip if the user changes
     condition_new_user = spts_tpls["user_id"] != spts_tpls["user_id"].shift(1)
 
     # start new trip if there is a new activity
     condition_new_activity = spts_tpls["last_activity"]
 
     # gap conditions
+    # start new trip after a gap
     gap = (spts_tpls["started_at_next"] - spts_tpls["finished_at"]) > gap_threshold
     condition_time_gap = gap.shift(1, fill_value=False)  # trip starts on next entry
-    # no gaps between activities
+
+    # no gaps between activities #todo: please elaborate condition further
```

Comment:
^

Reply:
It is an artifact from the time there where still intermediate activities. I think these aren't necessary anymore.